src/execution_plan/ops/shared/print_functions.c

```diff
@@ -7,23 +7,18 @@
 #include "print_functions.h"
 #include "../../execution_plan.h"
 
-static inline int _NodeToString(char *buf, uint buf_len, const char *alias, const char *label) {
+static inline void _NodeToString(sds *buf, const char *alias, const char *label) {
 	// Print a representation of a node with an optional alias and optional label.
-	int offset = snprintf(buf, buf_len, "(");
-	if(alias) offset += snprintf(buf + offset, buf_len - offset, "%s", alias);
-	if(label) offset += snprintf(buf + offset, buf_len - offset, ":%s", label);
-	offset += snprintf(buf + offset, buf_len - offset, ")");
-	return offset;
+	*buf = sdscatprintf(*buf, "(");
+	if(alias) *buf = sdscatprintf(*buf, "%s", alias);
+	if(label) *buf = sdscatprintf(*buf, ":%s", label);
+	*buf = sdscatprintf(*buf, ")");
 }
 
-int TraversalToString(const OpBase *op, char *buf, uint buf_len, AlgebraicExpression *ae) {
-	uint offset = 0;
-	if(!ae) {
-		offset += snprintf(buf, buf_len, "%s", op->name);
-		return offset;
-	}
+void TraversalToString(const OpBase *op, sds *buf, AlgebraicExpression *ae) {
+	if(!ae) *buf = sdscatprintf(*buf, "%s", op->name);
```

Comment:
I'm not sure about this line, was it here previously? the `if` condition doesn't make sense

Reply:
Yes it was here